src/lib/PostListCard.svelte

```diff
@@ -12,6 +23,25 @@
     day: 'numeric',
     year: 'numeric',
   })
+
+  export const FaIcons = {
+    tournament: faTrophy,
+    learning: faGraduationCap,
+    social: faComments,
+    awards: faAward,
+    charity: faHeart,
+    seasonal: faTree,
+  }
+
+  function isPostEventType(type: any): type is PostEventType {
+    return ['tournament', 'learning', 'social', 'awards', 'charity', 'seasonal'].includes(type)
+  }
+
+  const postEventType: PostEventType | undefined = isPostEventType(post.type)
+    ? post.type
+    : undefined
+  const faIcon = postEventType ? FaIcons[postEventType] : null
```

Comment:
You can get rid of all the complex logic, typing consideration, type guard etc. by just using your FaIcons object here directly:

```suggestion
  const faIcon = FaIcons[postEventType];
```

If you strictly need null as a value and not undefined just:

```suggestion
  const faIcon = FaIcons[postEventType] ?? null;
```

It shouldn't cause any typing issues, and if it does it should be easily resolvable.

Reply:
So this was catching whether or not the postEventType was present. Without null I would have every Post Page have an "undefined" postEventType display no matter what. If anything I agree that the second option seems good but I did get a type error. I will need to investigate further to see if I can get this to work.